include/exadg/incompressible_navier_stokes/postprocessor/postprocessor.cpp

```diff
@@ -195,116 +237,142 @@ PostProcessor<dim, Number>::do_postprocessing(VectorType const &     velocity,
 
 template<int dim, typename Number>
 void
-PostProcessor<dim, Number>::initialize_additional_vectors()
+PostProcessor<dim, Number>::initialize_derived_fields()
 {
   // vorticity
   if(pp_data.output_data.write_vorticity || pp_data.output_data.write_streamfunction ||
      pp_data.output_data.write_vorticity_magnitude)
   {
-    vorticity.type        = SolutionFieldType::vector;
-    vorticity.name        = "vorticity";
-    vorticity.dof_handler = &navier_stokes_operator->get_dof_handler_u();
-    navier_stokes_operator->initialize_vector_velocity(vorticity.get_vector_reference());
+    vorticity.type              = SolutionFieldType::vector;
+    vorticity.name              = "vorticity";
+    vorticity.dof_handler       = &navier_stokes_operator->get_dof_handler_u();
+    vorticity.initialize_vector = [&](std::shared_ptr<VectorType> & dst) {
```

Comment:
Initializing the vector by passing a lambda is not very intuitive in my opinion, but now it is not possible to abuse the ```get_reference_vector``` getter anymore, so I am OK with this. 

What I don't get is why the shared_ptr is introduced here. I think this should not be necessary.

Reply:
I removed it again here 655b8f49f28c45ea520cf681d308c7edc8801b99. It was an intermediate solution because of the mean_velocity implementation.

Regarding the lambda's, I want to do it consistently, so I see a strong analogy between initializing and recomputing the vector. Before, we needed a function `get_vector_reference()`, which I would like to avoid.

Reply:
Looks good to me :)